compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -6259,9 +6278,9 @@ private BType getQueryMapConstraintType(BType type, Location pos) {
                 return arrayType.eType;
             }
         } else if (type.tag == TypeTags.TUPLE) {
-            List<BType> tupleTypeList = ((BTupleType) type).tupleTypes;
-            if (tupleTypeList.size() == 2 && types.isAssignable(tupleTypeList.get(0), symTable.stringType)) {
-                return tupleTypeList.get(1);
+            List<BTupleMember> tupleTypeList = ((BTupleType) type).members;
```

Comment:
Why aren't we retrieving the list of types instead?

Reply:
We need to access `members` only when we need the entire member. If not, we can work with member types? Please recheck everywhere. Let's merge this PR once the other major concerns are sorted and fix things like this via another PR before the feature is merged to master.